tsdb/engine/tsm1/compact.go

```diff
@@ -777,13 +782,36 @@ func (c *Compactor) CompactFast(tsmFiles []string) ([]string, error) {
 	c.mu.RUnlock()
 
 	if !enabled {
+		c.mu.Lock()
+		defer c.mu.Unlock()
+		if err := c.cleanupDisabledCompaction(tsmFiles); err != nil {
+			return nil, err
+		}
 		return nil, errCompactionsDisabled
 	}
 
 	return files, err
 
 }
 
+// cleanupDisabledCompaction is responsible for cleaning up a compaction that
+// was started, but then abandoned before the temporary files were dealt with.
+func (c *Compactor) cleanupDisabledCompaction(tsmFiles []string) error {
```

Comment:
This looks like the older version.  The one in master has `remoteTempFiles` and doesn't do the globbing.